include/structured/Dialect/Indexing/IR/IndexingOps.td

```diff
@@ -29,16 +29,21 @@ def Indexing_GatherOp : Indexing_Op<"gather", [
   let arguments = (ins AnyRankedTensor:$source,
                        RankedTensorOf<[AnySignlessIntegerOrIndex]>:$indices,
                        DenseI64ArrayAttr:$gather_dims,
-                       UnitAttr:$unique);
+                       UnitAttr:$unique,
+                       UnitAttr:$is_slice);
```

Comment:
This is quite dangerous to implement as an attribute and it should be very easy to write a proper helper to detect that your indices come from some arange.